### PR TITLE
fix: Add boolean to CSS Prop type

### LINKED ISSUE #489: Conditional array CSS prop type errors
**Describe the bug**

The following code got TypeScript error.
https://compiledcssinjs.com/docs/css-prop#conditional-rules

```
TS2322: Type 'false | { border: string; color: string; }' is not assignable to type 'string | number | void | CSSProps | AnyKeyCssProps<void>'.   Type 'boolean' is not assignable to type 'string | number | void | CSSProps | AnyKeyCssProps<void>'.
```

**Expected behavior**
No errors.

**Screenshots**
error

**Desktop (please complete the following information):**

- OS: macOS BigSir 11.1
- Browser Google Chrome
- Version 87.0.4280.141（Official Build） （x86_64）

**Develop Environment**

- TypeScript: 4.1.3
- @compiled/react: 0.5.2

### PR DESCRIPTION
**Pull request by sonatard**
fix #489

This is my first contribution to JavaScript library 😄 

Please give me advice about tests and Storybook. I can't find CSS Prop tests.

### CODE REVIEW

**Comment by madou** [comment]
Cool! Good job

You can add a unit test here https://github.com/atlassian-labs/compiled/blob/master/packages/react/src/jsx/__tests__/index.test.tsx

You can do something similar to this test https://github.com/atlassian-labs/compiled/blob/master/packages/react/src/jsx/__tests__/index.test.tsx#L39

**Comment by pgmanutd** [comment]
Congrats @sonatard for your first contribution 🎉. See if you can add one boolean example using css array prop here: https://github.com/atlassian-labs/compiled/blob/master/examples/conditional-rules.tsx

**Comment by sonatard** [comment]
Thanks!! I'll try it.

**Comment by madou** [comment]
No worries - let us know how you go! Were you able to run storybook and tests locally? Anything that could be documented better in CONTRIBUTING.md?

**Comment by sonatard** [comment]
>  Were you able to run storybook and tests locally? 

Yes, No Problem!
I added tests and Storybook. Please review it.

And It ’s a little off topic, I use WebStorm which prints `Attribute css is not allowed here` warning in test files.
Test files import `import '@compiled/react';`. I can't check type error in test  file and Storybook.

スクリーンショット 2021-01-12 10 27 10

**Comment by madou** [comment]
Thanks! Will review soon. 

For the webstorm error - from looking at the error I think you need to just add it to webstorm as a custom HTML attribute? It must have its own linting 🤔

File: packages/react/src/jsx/__tests__/index.test.tsx [hunk 1]
```diff
@@ -129,4 +129,31 @@ describe('css prop', () => {
...
+
+    expect(getByText('hello world')).toHaveCompiledCss('font-size', '12px');
+  });
+
+  it('should create css from object reference with false condition', () => {
+    const condition = false;
+    const base = { fontSize: 12 };
+    const { getByText } = render(<div css={condition && base}>hello world</div>);
+
+    expect(getByText('hello world')).not.toHaveCompiledCss('font-size', '12px');
```
Comment (sonatard) [inline packages/react/src/jsx/__tests__/index.test.tsx:146]: I want to use an empty expect value. But I don't know how to use it.

Comment (madou) [reply packages/react/src/jsx/__tests__/index.test.tsx:146]: hmmm that doesn't exist ATM

I think your assertion is okay though - it _is_ asserting the expected behaviour 😄

File: examples/static-evaluation.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { styled } from '@compiled/react';
+import '@compiled/react';
```
Comment (madou) [inline examples/static-evaluation.tsx:2]: nit: this isn't needed - css prop is enabled when the module is found so the above `@compiled/react` has enabled it 👍

Comment (sonatard) [reply examples/static-evaluation.tsx]: I don't know why I added this. 🤣

**Review by madou — APPROVED** [review]
LGTM! I'll just pull your fork locally to make sure the VR tests look good - unfortunately they aren't being ran in forked pull requests 😭 (made an issue here #497)

**Comment by sonatard** [comment]
Thanks for reviewing!! Is there anything else I need to do?

**Comment by madou** [comment]
Not at the moment, thanks!

**Pull request merged**